powerauth-push-server/src/main/java/io/getlime/push/service/fcm/FcmClient.java

```diff
@@ -16,82 +16,109 @@
 
 package io.getlime.push.service.fcm;
 
+import io.getlime.push.configuration.PushServiceConfiguration;
 import io.getlime.push.service.fcm.model.FcmSendRequest;
 import io.getlime.push.service.fcm.model.FcmSendResponse;
-import org.apache.http.HttpHost;
-import org.apache.http.auth.AuthScope;
-import org.apache.http.auth.UsernamePasswordCredentials;
-import org.apache.http.client.CredentialsProvider;
-import org.apache.http.impl.client.BasicCredentialsProvider;
-import org.apache.http.impl.client.ProxyAuthenticationStrategy;
-import org.apache.http.impl.nio.client.CloseableHttpAsyncClient;
-import org.apache.http.impl.nio.client.HttpAsyncClientBuilder;
-import org.springframework.http.*;
-import org.springframework.http.client.HttpComponentsAsyncClientHttpRequestFactory;
-import org.springframework.util.concurrent.ListenableFuture;
-import org.springframework.web.client.AsyncRestTemplate;
-import org.springframework.web.client.RestClientException;
+import io.netty.channel.ChannelOption;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.http.client.reactive.ClientHttpConnector;
+import org.springframework.http.client.reactive.ReactorClientHttpConnector;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.BodyInserters;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.ipc.netty.http.client.HttpClientOptions;
+import reactor.ipc.netty.options.ClientProxyOptions;
+
+import java.util.function.Consumer;
 
 /**
  * FCM server client
  *
  * @author Petr Dvorak, petr@lime-company.eu
  */
+@Service
```

Comment:
Just a quick note before it is finished: this is completely wrong. We need to construct multiple instance of `FcmClient`, one for each application that is configured in the system (with credentials that are specific for that app). This is why `prepareFcmClient` uses `new` keyword to create a new instance...

Reply:
I see, ok, I will revert the change and handle autowiring of configuration otherwise.